source/common/io/io_uring_impl.cc

```diff
@@ -0,0 +1,138 @@
+#include "source/common/io/io_uring_impl.h"
+
+#include <sys/eventfd.h>
+
+#include "source/common/common/utility.h"
+
+namespace Envoy {
+namespace Io {
+
+thread_local bool IoUringFactoryImpl::is_instantiated_ = false;
+
+IoUringFactoryImpl::IoUringFactoryImpl(uint32_t io_uring_size, bool use_submission_queue_polling)
```

Comment:
This is interesting. I'll try that.

Reply:
I've made the factory to use Envoy's thread local storage. It works for this PR. But then I ported the change to the [draft PR](https://github.com/envoyproxy/envoy/pull/19082) with io_uring-backed iohandle and its integration test started to fail because the client connection created in the test_thread wants to use IoUring too. test_thread is not registered with TLS -> its slot is not set -> crash. Can test_thread be registered properly?

Also what's the benefit of using Envoy's TLS here?

Reply:
I turned the extension back to core per Harvey's comment. Yet Envoy's TLS is used now instead of `thread_local`.